compiler/noirc_frontend/src/elaborator/expressions.rs

```diff
@@ -369,7 +369,7 @@ impl<'context> Elaborator<'context> {
                     function_args.push((typ, arg, span));
                 }
 
-                let location = Location::new(span, self.file);
+                let location = Location::new(method_name_span, self.file);
```

Comment:
I changed this because...

Given a call like this:

```rust
foo.bar(1);
```

two ExprId are interned (their locations and types):
1. The method call function type (for example `fn(i32) -> Field` assuming `bar` returns `Field`)
2. The expression's type (`Field`)

Both used the same span so later when getting the type at that location we'd get two types and only one of them would win.

But I think it's more correct to only use the location of the method's name (bar) here if we are going to intern the function type, and not the return type. But let me know if this change is wrong!

Reply:
This is technically the location for the entire method call though and not just the method name. Was this causing issues for the LSP?

Reply:
Ah, good point. I changed it to be the span from the object to the name, like this:

```rust
foo.bar(...)
^^^^^^^
```

It was causing issues because in `id_to_location` and `id_to_type` we captured two expressions with the same location and different types (one for the function type, another for the function return type). LSP relies on going from a location to an expression to a type, and it was finding two expressions in that locations. The logic in that case is to use the first one, but they are unordered so we could get the correct one or the wrong one.

I'm not sure why the function type is captured in `id_to_location` and `id_to_type`, though, but maybe if the type i the function type maybe it makes more sense for its location to not include the arguments, etc.